src/test/java/org/openrewrite/java/dependencies/ChangeDependencyTest.java

```diff
@@ -146,4 +146,44 @@ void doNotPinWhenNotVersioned() {
               """)
         );
     }
+
+    @Test
+    void pinWhenOverrideManagedVersionGradle() {
+        rewriteRun(
+          spec -> spec
+            .beforeRecipe(withToolingApi())
+            .recipe(new ChangeDependency("mysql", "mysql-connector-java", "com.mysql", "mysql-connector-j", "8.0.x", null, true)),
+          buildGradle(
+            """
+              plugins {
+                id 'java'
+                id 'org.springframework.boot' version '2.6.1'
+                id 'io.spring.dependency-management' version '1.0.11.RELEASE'
+              }
+              
+              repositories {
+                 mavenCentral()
+              }
+              
+              dependencies {
+                  runtimeOnly 'mysql:mysql-connector-java'
+              }
+              """,
+            """
+              plugins {
+                id 'java'
+                id 'org.springframework.boot' version '2.6.1'
+                id 'io.spring.dependency-management' version '1.0.11.RELEASE'
+              }
+              
+              repositories {
+                 mavenCentral()
+              }
+              
+              dependencies {
+                  runtimeOnly 'com.mysql:mysql-connector-j:8.0.33'
```

Comment:
Won't this test need to be updated with every new release of mysql-connector?
It slightly feels like we're re-testing something that's covered elsewhere already.
Perhaps we can lose this test and just rely on the tests in openrewrite/rewrite.

Reply:
I copied this test from rewrite-spring, and I think the selector is intentionally 8.0.x because there is a new 8.1.0 release, and probably the 8.0.x branch won't have more releases?
I agree that we are already testing this in rewrite, but here the idea was to use the new constructor that supports this "overwrite" that previously was not possible without it for maven, so now, this "composite recipe" also supports it.